src/linked-list/iterator.ts

```diff
@@ -0,0 +1,23 @@
+import {ADTLinkedList} from '../linked-list';
+import {ADTLinkedListElement as Element} from '../linked-list/element';
+import {IterableType} from '../iterable-type';
+
+export class LinkedListIterator<ItemT> implements Iterator<ItemT | null> {
+	private item: Element<ItemT> | null;
+
+	constructor(linkedList: ADTLinkedList<ItemT>) {
+		this.item = linkedList.head();
+	}
+
+	next(): IterableType<ItemT | null> {
+		if (this.item == null) {
```

Comment:
- [ ] Swap this double equality `null` check for an if falsey check `if (!this.item) {`
Toreda TypeScript standards require all comparisons use triple equality `===` checks when testing values. This `==` check was likely used because `this.item == null` catches both `null` and `undefined`. The check itself is not bad/wrong, but has a high probability to become a bug over time if `==` is ever changed to `===`, leading to the edge case where undefined is no longer caught.

A stronger guarantee here is inverting the check to see if its falsey and early drop if so.